### ISSUE TITLE
Launchdarkly_project apply doesn't update the nested environments

### ISSUE THREAD

**Issue by hsy3418** [launchdarkly/terraform-provider-launchdarkly#50]
I am trying to update nested environments for an existing project by removing one environment.
Using Terraform v0.13.5 to do  `terraform plan`, it shows the changes as expected, and then `terraform apply` , it shows `Apply complete`, but the environment is not really removed from the UI, but apparently the terraform state file has been updated with the expected one, so has to manually removed from the UI. Here are the success terraform apply results
```
Terraform will perform the following actions:
  # launchdarkly_project.example-project will be updated in-place
  ~ resource "launchdarkly_project" "example-project" {
        id                 = "example-project"
        include_in_snippet = false
        key                = "example-project"
        name               = "Example project"
        tags               = [
            "terraform",
        ]
        environments {
            api_key              = (sensitive value)
            client_side_id       = (sensitive value)
            color                = "EEEEEE"
            confirm_changes      = false
            default_track_events = false
            default_ttl          = 0
            key                  = "production"
            mobile_key           = (sensitive value)
            name                 = "Production"
            require_comments     = false
            secure_mode          = false
            tags                 = [
                "terraform",
            ]
        }
      - environments {
          - api_key              = (sensitive value)
          - client_side_id       = (sensitive value)
          - color                = "000000" -> null
          - confirm_changes      = false -> null
          - default_track_events = false -> null
          - default_ttl          = 0 -> null
          - key                  = "staging" -> null
          - mobile_key           = (sensitive value)
          - name                 = "Staging" -> null
          - require_comments     = false -> null
          - secure_mode          = false -> null
          - tags                 = [
              - "terraform",
            ] -> null
        }
    }
Plan: 0 to add, 1 to change, 0 to destroy.
Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.
  Enter a value: yes
launchdarkly_project.example-project: Modifying... [id=example-project]
launchdarkly_project.example-project: Modifications complete after 1s [id=example-project]
Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

**Comment by ldhenry** [comment]
Hey @hsy3418,

Thanks for bringing this to our attention. I've created an internal ticket to track this issue and I'll be sure to report back here once we have a fix.

Thanks,
Henry

**Comment by ldhenry** [comment]
Hi @hsy3418,

Sorry for the long delay on this issue. It turned out to be a tricky one. Unfortunately, we will not be able to add support for deleting nested environments via our Terraform provider in 1.X releases because prior to v1.5, environments were "computed" in the project resource. When we made the change to support optional nested environments in v1.5, we included the following note in our changelog:
> The launchdarkly_project's environments attribute is no longer marked as computed. This means that if you have launchdarkly_project resources without nested environments that were created before this version, you will see a diff denoting the removal of the computed environments from your state. It is safe to apply this change as no changes be made to your LaunchDarkly resources when applied.

If we were to support the deletion of nested environments in a 1.X release and a customer upgraded from a release before v1.5, their environments would be unexpectedly deleted, which could lead to major problems for our customers.

The good news is that we do plan to support deleting nested environments in v2.0 and we are aiming to make v2.0 our next release. As it is a major version release, we will be taking more calculated and cautious approach in order prevent bugs like this from cropping up in the future. While I can't promise a rapid resolution to this issue, I will keep it open until it is fixed in v2.0 and I will let you know here when the release is available.

In the meantime, if you need to delete environments using Terraform, using the dedicated `launchdarkly_environment` resource will work - though you will not be able to delete the default "Test" and "Projection" environments that are created by the `launchdarkly_project` resource with doing a `terraform import` first.

Thanks,
Henry

**Comment by hsy3418** [comment]
Thanks Henry for the reply!

**Comment by sloloris** [comment]
Hi @hsy3418!

We're pleased to let you know that we have released v2.0.0 of our provider with a fix for this very issue! Starting with v2, we now require you configure environments on your project resources and recommend that you no longer use the environment resource unless you wish to configure environments for projects not also managed by Terraform.

Since this is a major version upgrade, you will likely need to make changes to some of your configurations, and we intend to release an upgrade guide for this soon on our [docs website](https://docs.launchdarkly.com/integrations/terraform).
